### PR TITLE
Cms-Kit page system implementation 

### LINKED ISSUE #6847: CMS Kit - Page System Implementation

### PR DESCRIPTION
**Pull request by cotur**
Closes: #6847

### CODE REVIEW

**Review by enisn — APPROVED** [review]

File: modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Pages/PageAdminController.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        [HttpDelete]
+        [Authorize(CmsKitAdminPermissions.Pages.Delete)]
+        [Route("{id}")]
+        public virtual Task DeleteAsync(Guid id)
+        {
+            return PageAdminAppService.DeleteAsync(id);
+        }
+
+        [HttpGet]
+        [Route("exists/{url}")]
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Admin.HttpApi/Volo/CmsKit/Admin/Pages/PageAdminController.cs]: We should use HTTP Head to check the existence of the URL. This implementation feels like an RPC. We should do it like HTTP Head `api/cms-kit-admin/pages\{url}`

File: modules/cms-kit/test/Volo.CmsKit.Application.Tests/Pages/PageAdminAppService_Tests.cs [hunk 2]
```diff
@@ -38,39 +66,41 @@ public async Task ShouldCreateAsync()
...
             var dto = new CreatePageInputDto
             {
                 Title = "test",
-                Url = _data.Page_1_Url
+                Url = _data.Page_1_Url,
+                Content = "test-content"
             };
 
-            await Should.ThrowAsync<Exception>(async () => await _pageAdminAppService.CreatePageAsync(dto));
+            await Should.ThrowAsync<Exception>(async () => await _pageAdminAppService.CreateAsync(dto));
```
Comment (ilkayilknur) [inline modules/cms-kit/test/Volo.CmsKit.Application.Tests/Pages/PageAdminAppService_Tests.cs]: We should create a spesific business exception here. What kind of exception does this call throw?

**Review by ilkayilknur — CHANGES_REQUESTED** [review]

File: modules/cms-kit/src/Volo.CmsKit.Domain.Shared/Volo/CmsKit/Localization/Resources/en.json [hunk 3]
```diff
@@ -30,6 +34,7 @@
     "Undo": "Undo",
     "Update": "Update",
     "YourComment": "Your comment",
-    "YourReply": "Your reply"
+    "YourReply": "Your reply",
+    "CmsKit:Page:0001": "The given url ({0}) is already exist."
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.Domain.Shared/Volo/CmsKit/Localization/Resources/en.json]: The given url ({0}) already exists.

File: modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Pages/MongoPageRepository.cs [hunk 4]
```diff
@@ -13,19 +18,48 @@ public MongoPageRepository(IMongoDbContextProvider<ICmsKitMongoDbContext> dbCont
         {
         }
 
-        public Task<Page> GetByUrlAsync(string url)
+        public virtual async Task<int> GetCountAsync(string filter = null, CancellationToken cancellationToken = default)
         {
-            return GetAsync(x => x.Url == url);
+            return await (await GetMongoQueryableAsync(GetCancellationToken(cancellationToken)))
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Pages/MongoPageRepository.cs]: We can call `GetCancellationToken(cancellationToken)` once use returning value everywhere.

File: modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Pages/MongoPageRepository.cs [hunk 5]
```diff
@@ -13,19 +18,48 @@ public MongoPageRepository(IMongoDbContextProvider<ICmsKitMongoDbContext> dbCont
...
+                .WhereIf<Page, IMongoQueryable<Page>>(
+                    !filter.IsNullOrWhiteSpace(),
+                    u =>
+                        u.Title.Contains(filter)
+                )
+                .OrderBy(sorting ?? nameof(Page.Title))
+                .As<IMongoQueryable<Page>>()
+                .PageBy<Page, IMongoQueryable<Page>>(skipCount, maxResultCount)
+                .ToListAsync(GetCancellationToken(cancellationToken));
```
Comment (ilkayilknur) [inline modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Pages/MongoPageRepository.cs]: Please see my previous comment.

**Review by ilkayilknur — CHANGES_REQUESTED** [review]
I think we can also remove the `Description` property from everywhere after our last meeting. Rest LGTM.

**Review by ilkayilknur — APPROVED** [review]

**Pull request merged**